backend/fms_core/viewsets/study.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import viewsets
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from fms_core.models import Study
+from fms_core.serializers import StudySerializer
+from fms_core.services.study import create_study
+from django.core.exceptions import ValidationError
+
+import string
+
+from ._constants import _study_filterset_fields
+
+class StudyViewSet(viewsets.ModelViewSet):
+    queryset = Study.objects.all()
+    serializer_class = StudySerializer
+    permission_classes = [IsAuthenticated]
+
+    filterset_fields = {
+        **_study_filterset_fields,
+    }
+
+    def create(self, request, *args, **kwargs):
+        study_data = request.data
+
+        # Generate a sequential letter by counting the number of existing studies tied to the provided project
+        study_count = Study.objects.filter(project=study_data['project']).count()
+        letter = string.ascii_uppercase[study_count]
+
+        # Call create study service
+        study, errors, warnings = create_study(letter=letter,
```

Comment:
The frontend sends the project ID in the `project` field (and a workflow ID in the `workflow` field). When those are passed to `create_study` I get an exception because the service is expecting Project and Workflow objects, not ID's.

Reply:
Forget what i told you earlier i thought you were hitting the default django create endpoint. If you are hitting the create added yesterday ... it's a work in progress. we'll have to fetch to either fetch the objects using the ids or change the service to take directly ids. Do not change your call on behalf of this. I'll tweak the code some so it works.